src/main/java/bio/terra/app/model/CloudRegion.java

```diff
@@ -0,0 +1,71 @@
+package bio.terra.app.model;
+
+
+import bio.terra.common.EnumUtils;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+
+import java.io.IOException;
+
+/**
+ * Represents regions across different cloud providers
+ */
+@JsonDeserialize(using = CloudRegion.CloudRegionDeserializer.class)
+public interface CloudRegion {
+
+    /**
+     * Returns the implementing enumeration name
+     */
+    String name();
+
+    /**
+     * Return the region string that the cloud provider can understand
+     */
+    String getValue();
+
+    class CloudRegionDeserializer extends StdDeserializer<CloudRegion> {
+
+        public CloudRegionDeserializer() {
+            this(CloudRegion.class);
+        }
+        public CloudRegionDeserializer(Class<?> vc) {
+            super(vc);
+        }
+
+        @Override
+        public CloudRegion deserialize(
```

Comment:
Have you heard of jackson support for deduction based polymorphism? I read about a while back, maybe it can help here? https://cowtowncoder.medium.com/jackson-2-12-most-wanted-1-5-deduction-based-polymorphism-c7fb51db7818

Reply:
I got the current approach from the docs but I'll look here too...might be a cleaner way to do it